lib/Mojo/IOLoop/Subprocess.pm

```diff
@@ -55,6 +62,32 @@ sub _start {
       $self->$parent(shift(@$results) // $@, @$results);
     }
   );
+  my $progress_stream = Mojo::IOLoop::Stream->new($progress_reader)->timeout(0);
+  $self->ioloop->stream($progress_stream);
+  my $progress_buffer = '';
+  $progress_stream->on(
+    read => sub {
+      $progress_buffer .= pop;
+      while (1) {
+        my ($length) = $progress_buffer =~ /\A(\d+)\-/;
+        defined $length or last;
+        my $snippet_size = length($length) + 1 + $length;
+        length($progress_buffer) >= $snippet_size or last;
+        my $snippet = substr $progress_buffer, 0, $snippet_size, '';
+        $snippet =~ s/\A$length\-//;
+        my $args = $self->deserialize->($snippet);
+        $self->emit(progress => @$args);
+      }
```

Comment:
This block is in need of some golfing, it seems way too verbose. 😉

Reply:
`pos` can replace the `length($length) + 1`
```perl
        my ($length, $s) = (($progress_buffer =~ m/\A(\d+)\-/gc), pos($progress_buffer));
        defined $length or last;
        length($prog_buffer) >= ($length += $s) or last;
        my $args = $self->deserialize->(substr substr($progress_buffer, 0, $length, ''), $s);
        $self->emit(progress => @$args);
```